avocado/utils/disk.py

```diff
@@ -121,15 +121,28 @@ def get_disks():
     :returns: a list of paths to the physical disks on the system
     :rtype: list of str
     """
-    json_result = process.run("lsblk --json --paths --inverse")
-    json_data = json.loads(json_result.stdout_text)
-    disks = []
-    for device in json_data["blockdevices"]:
-        disks.append(device["name"])
-        if "children" in device:
-            for child in device["children"]:
-                disks.append(child["name"])
-    return disks
+    try:
+        json_result = process.run("lsblk --json --paths --inverse")
+        json_data = json.loads(json_result.stdout_text)
+        disks = []
+        for device in json_data["blockdevices"]:
+            disks.append(device["name"])
+            if "children" in device:
+                for child in device["children"]:
+                    disks.append(child["name"])
+        return disks
+    except subprocess.CalledProcessError as pe:
```

Comment:
Avocado's `avocado.utils.process.run()` will raise `avocado.utils.process.CmdError` instead.  Also, consider moving the `except` block to immediately after the `process.run()` call, and add a second for `try/except` block for the JSON parsing code.

Reply:
Thank you very much for your suggestion, please help to have a look again